iocage/lib/Jail.py

```diff
@@ -829,14 +827,27 @@ def _jail_conf_file(self) -> str:
     def restart(
         self,
         shutdown: bool=False,
-        force: bool=False
+        force: bool=False,
+        event_scope: typing.Optional['iocage.lib.events.Scope']=None
     ) -> typing.Generator['iocage.lib.events.IocageEvent', None, None]:
         """Restart the jail."""
         failed: bool = False
-        jailRestartEvent = iocage.lib.events.JailRestart(jail=self)
-        jailShutdownEvent = iocage.lib.events.JailShutdown(jail=self)
-        JailSoftShutdownEvent = iocage.lib.events.JailSoftShutdown(jail=self)
-        jailStartEvent = iocage.lib.events.JailStart(jail=self)
+        jailRestartEvent = iocage.lib.events.JailRestart(
+            jail=self,
+            scope=event_scope
+        )
+        jailShutdownEvent = iocage.lib.events.JailShutdown(
+            jail=self,
+            scope=jailRestartEvent.scope
+        )
+        JailSoftShutdownEvent = iocage.lib.events.JailSoftShutdown(
+            jail=self,
+            scope=jailRestartEvent.scope
+        )
+        jailStartEvent = iocage.lib.events.JailStart(
+            jail=self,
+            scope=jailRestartEvent.scope
```

Comment:
Because a new scope is needed on any new action. When you start a jail, this is the entry point where you can leave the scope unspecified. All successors invoked by this operation will use the same scope. There is one occasion where the stop() method is called right after a start. Both belong to the same history scope, but might also be invoked independently.

The API does not change and keeps the "old" behavior. Just that command invocation does not pollute a global history - it exists only for the duration of such actions.

Reply:
so in that sense, we're not creating a strong coupling between all our classes and EventScope, but only between all our events.

Reply:
yes, but within the other classes. At least I can say that being explicit is better than implicit 
- probably this all needs to be written down in documentation.